chore: AT-TLS context configuration tests

pablocarle opened the pull request:
# Description

Add tests for context configuration of AT-TLS in the core services, this verifies the configuration is correct from the Spring point of view, it does not test the AT-TLS context itself.

Linked to #2795 

## Type of change

Please delete options that are not relevant.

- [ ] chore: Chore, repository cleanup, updates the dependencies.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] PR title conforms to commit message guideline ## [Commit Message Structure Guideline](../CONTRIBUTING.md)
- [ ] I have commented my code, particularly in hard-to-understand areas. In JS I did provide JSDoc
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The java tests in the area I was working on leverage @Nested annotations
- [ ] Any dependent changes have been merged and published in downstream modules

For more details about how should the code look like read the [Contributing guideline](https://github.com/zowe/api-layer/blob/master/CONTRIBUTING.md)

Commit:
add tests (disabled) plus settings updates

Signed-off-by: Pablo Hernán Carle <pablo.carle@broadcom.com>

Commit:
Merge remote-tracking branch 'origin/v2.x.x' into reboot/attls-test

Commit:
Merge remote-tracking branch 'origin/v2.x.x' into reboot/attls-test

Commit:
wip tests in gateway and discovery for attls

Signed-off-by: Pablo Hernán Carle <pablo.carle@broadcom.com>

pablocarle committed:
Merge branch 'v2.x.x' into reboot/attls-test

balhar-jakub commented on gateway-service/src/test/java/org/zowe/apiml/gateway/security/config/AttlsConfigTest.java:
```diff
@@ -0,0 +1,86 @@
+    class GivenAttlsModeEnabled {
+
+        @Test
+        void whenContextLoads_RequestFailsWithHttps() {
```
You can nest the tests into WhenContextLoads nested class.

balhar-jakub commented on discovery-service/src/test/java/org/zowe/apiml/discovery/config/AttlsConfigTest.java:
```diff
@@ -0,0 +1,93 @@
+        }
+
+        @Test
+        void whenContextLoads_RequestFailsWithHttps() {
```
Minor Note you can nest via: 

@Nested
class WhenContextFalls
  @Nested 
  class GivenAttlsModeEnabledAndHttps
  @Nested
  class GivenAttlsModeEnabledAndHttp

pablocarle replied:
done, thank you

Commit:
address review, add tests in caching and catalog

Signed-off-by: Pablo Hernán Carle <pablo.carle@broadcom.com>

Commit:
update port

Signed-off-by: Pablo Hernán Carle <pablo.carle@broadcom.com>

Commit:
fix gateway acceptance test

Signed-off-by: Pablo Hernán Carle <pablo.carle@broadcom.com>

pablocarle committed:
Merge branch 'v2.x.x' into reboot/attls-test

Commit:
fix test

Signed-off-by: Pablo Hernán Carle <pablo.carle@broadcom.com>

Commit:
Merge branch 'reboot/attls-test' of https://github.com/zowe/api-layer into reboot/attls-test

achmelo commented on .gitignore:
```diff
@@ -119,3 +119,6 @@ onboarding-enabler-micronaut/out/
 
 file/
+
+# zowe-cli-id-federation-plugin
```
I would remove this comment

achmelo commented on api-catalog-services/src/test/java/org/zowe/apiml/apicatalog/functional/ApiCatalogFunctionalTest.java:
```diff
@@ -33,6 +37,10 @@ void setUp() {
+    }
+
+    protected String getCatalogUriWithPath(String scheme, String path) {
+        return String.format("%s://%s:%d", scheme, hostname, port) + path;
```
is it possible to use path as another parameter in format()?

pablocarle replied:
done

achmelo commented on caching-service/src/test/java/org/zowe/apiml/caching/config/AttlsConfigTest.java:
```diff
@@ -0,0 +1,90 @@
+    class GivenAttlsModeEnabled {
+
+        private String getUri(String scheme, String endpoint) {
+            return String.format("%s://%s:%s%s%s", scheme, hostname, port, "/api/v1", endpoint);
```
`return String.format("%s://%s:%d/%s/%s", scheme, hostname, port, "api/v1", endpoint)`

achmelo commented on gateway-service/src/test/java/org/zowe/apiml/acceptance/config/GatewayOverrideConfig.java:
```diff
@@ -78,7 +78,6 @@ public SimpleRouteLocator simpleRouteLocator() {
 
     @Bean
     public ApplicationRegistry registry() {
-
```
I would revert changes like this, it is more readable then

pablocarle replied:
do you mean this empty space?

achmelo replied:
yes, I am fine if you leave them here, it is just increasing the changes and (probably) not adding value

achmelo commented on gateway-service/src/test/java/org/zowe/apiml/gateway/security/config/AttlsConfigTest.java:
```diff
@@ -0,0 +1,89 @@
+                    .log().all()
+                    .cookie(COOKIE_AUTH_NAME, "jwttoken")
+                .when()
+                    .get(String.format("http://localhost:%d", port) + serviceWithDefaultConfiguration.getPath())
```
same as other String.format()

pablocarle replied:
done

Commit:
address review

Signed-off-by: Pablo Hernán Carle <pablo.carle@broadcom.com>

Commit:
remove ignore comment

Signed-off-by: Pablo Hernán Carle <pablo.carle@broadcom.com>

Commit:
update assertions

Signed-off-by: Pablo Hernán Carle <pablo.carle@broadcom.com>

Commit:
try to address failed test in sonar stage

Signed-off-by: Pablo Hernán Carle <pablo.carle@broadcom.com>

Commit:
skip constructor call in test

the test calls the real constructor which has an automatic java executor call, makes it likely flaky.

Signed-off-by: Pablo Hernán Carle <pablo.carle@broadcom.com>

Commit:
wip try fix test

Signed-off-by: Pablo Hernán Carle <pablo.carle@broadcom.com>

Commit:
wip

Signed-off-by: Pablo Hernán Carle <pablo.carle@broadcom.com>

Commit:
wip try replace executor

Signed-off-by: Pablo Hernán Carle <pablo.carle@broadcom.com>

Commit:
wip try fix test

Signed-off-by: Pablo Hernán Carle <pablo.carle@broadcom.com>
